lib/src/model/package.dart

```diff
@@ -181,7 +181,19 @@ class Package extends LibraryContainer
   @override
   String get enclosingName => packageGraph.defaultPackageName;
 
-  String get filePath => 'index.html';
+  String get filePath => 'index.$fileType';
+
+  String get fileType {
+    // TODO(jdkoren): Provide a way to determine file type of a remote package's
+    // docs. Perhaps make this configurable through dartdoc options.
+    // In theory, a remote package could be documented in any supported format.
+    // In practice, devs depend on Dart, Flutter, and/or packages fetched
+    // from pub.dev, and we know that all of those use html docs.
+    if (package.documentedWhere == DocumentLocation.remote) {
+      return 'html';
+    }
+    return packageGraph.rendererFactory.fileTypeRenderer.fileType;
```

Comment:
I'm contemplating using the `config` member to lookup the `format` instead of using a renderer here. That requires moving `format` out of `GeneratorContext` and into `DartdocOptionContext`.

Reply:
Switched to using the config member as it's a bit more direct.